Source/SuperLinq/Exclude.cs

```diff
@@ -46,50 +46,156 @@ public static IEnumerable<T> Exclude<T>(this IEnumerable<T> sequence, int startI
 		};
 	}
 
-	private sealed class ExcludeCollectionIterator<T>(
-		ICollection<T> source,
-		int startIndex,
-		int count
-	) : CollectionIterator<T>
+	/// <summary>
+	///	    Excludes a contiguous number of elements from a sequence starting at a given index.
+	/// </summary>
+	/// <typeparam name="T">
+	///	    The type of the elements of the sequence
+	/// </typeparam>
+	/// <param name="sequence">
+	///	    The sequence to exclude elements from
+	/// </param>
+	/// <param name="range">
+	///	    The zero-based index at which to begin excluding elements
+	/// </param>
+	/// <returns>
+	///	    A sequence that excludes the specified portion of elements
+	/// </returns>
+	/// <exception cref="ArgumentNullException">
+	///	    <paramref name="sequence"/> is <see langword="null" />.
+	/// </exception>
+	/// <exception cref="ArgumentOutOfRangeException">
+	///	    <paramref name="range.Start.Value"/> or <paramref name="range.End.Value"/> is less than <c>0</c>.
+	/// </exception>
+	/// <remarks>
+	///	    This method uses deferred execution and streams its results.
+	/// </remarks>
+	public static IEnumerable<T> Exclude<T>(this IEnumerable<T> sequence, Range range)
 	{
+		ArgumentNullException.ThrowIfNull(sequence);
+		ArgumentOutOfRangeException.ThrowIfNegative(range.Start.Value);
+		ArgumentOutOfRangeException.ThrowIfNegative(range.End.Value);
+
+		if (range.Start.Value - range.End.Value == 0)
+			return sequence;
+
+		return sequence switch
+		{
+			IList<T> list => new ExcludeListIterator<T>(list, range),
+			ICollection<T> collection => new ExcludeCollectionIterator<T>(collection, range),
+			_ => ExcludeCore(sequence, range)
+		};
```

Comment:
```suggestion
		ArgumentNullException.ThrowIfNull(sequence);

		if ((range.Start.IsFromEnd, range.End.IsFromEnd) == (false, false))
		{
			return Exclude(sequence, range.Start.Value, range.End.Value - range.Start.Value);
		}

		if (sequence.TryGetNonEnumeratedCount(out var count))
		{
			var (start, length) = range.GetOffsetAndLength(count);
			return Exclude(sequence, start, length);
		}

		return ExcludeCore(sequence, range);
```

Reply:
This will defer to existing processing in as many cases as possible; it also simplifies the `ExcludeCollectionIterator` and `ExcludeListIterator` back to their original forms, because they will only get called from the `int, int` version of `Exclude`.